src/spark_rapids_ml/core.py

```diff
@@ -358,7 +360,15 @@ def _enable_nccl(self) -> bool:
         """If enable or disable NCCL communication"""
         return True
 
-    def _fit(self, dataset: DataFrame) -> "_CumlModel":
+    def _require_ucx(self) -> bool:
+        """If enable or disable ucx over NCCL"""
+        return False
+
+    def _return_model(self) -> bool:
+        """If _fit returns a model or a RDD"""
+        return True
+
+    def _fit(self, dataset: DataFrame) -> Union["_CumlModel", RDD]:
```

Comment:
if you really want to return RDD. I'd like to suggest you move the code to a separate function to reuse some code instead of changing the "_fit"

Reply:
Focusing on KNN functionality for now. 

Created a ticket to improve coding style in the future.

Reply:
Could you move the common code in "_fit" into another function (eg, abc_func) to return the RDD. and in KNN please overwrite fit() which should call your abc_func instead of "_fit"

Reply:
Copied the latest comment into the ticket for record. Will do this in the next PRs to avoid this one being too long.

Reply:
@wbo4958 @lijinf2 will do this or similar  in a followup pr.